src/lib/controller/http-client.ts

```diff
@@ -0,0 +1,107 @@
+import { PUBLIC_API_BASE_URL } from "$env/static/public";
+
+/**
+ * A simple HTTP client that can make requests to the backend API.
+ *
+ * This class is used to make requests to the backend API. It is a simple wrapper around the `fetch` API.
+ * It uses the `PUBLIC_API_BASE_URL` environment variable to determine the base URL of the API.
+ *
+ * @example
+ * const client = new HttpClient("api/v1/example");
+ * await client.get();
+ * await client.post({ data: "example" });
+ * await client.put({ data: "example" });
+ * await client.patch({ data: "example" });
+ * await client.delete();
+ */
+export class HttpClient {
+    private basePath: string;
+
+    /**
+     * Creates a new HTTP client.
+     *
+     * @param path The path to the API endpoint. This path will be appended to the `PUBLIC_API_BASE_URL` environment variable.
+     *
+     * @example
+     * const client = new HttpClient("api/v1/example");
+     * // When the `PUBLIC_API_BASE_URL` environment variable is `http://localhost:3000`
+     * // the base URL of the API will be `http://localhost:3000/api/v1/example`
+     */
+    constructor(path: string = "") {
+        this.basePath = `${PUBLIC_API_BASE_URL}${path ? `/${path}` : ""}`;
```

Comment:
```suggestion
        this.basePath = `${PUBLIC_API_BASE_URL}${path ? `/${path}` : "/"}`;
```

Don't you need a forward slash in default case?

Reply:
Afaik, trailing slashes can usually be omitted.

Reply:
They can be omitted, especially since it is only the base path and all further relative paths starts with a slash, but I thought is is better practice to provide it (https://stackoverflow.com/questions/5948659/when-should-i-use-a-trailing-slash-in-my-url)

Reply:
I'm not sure what your reasoning is by relating to this article because they all argument that a trailing slash doesn't make sense in most cases. In the [OpenAPI Spec](https://spec.openapis.org/oas/v3.1.1.html#paths-object) trailing slashes are omitted.